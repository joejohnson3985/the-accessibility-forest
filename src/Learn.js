import React, { Component } from 'react';
import './Learn.scss';
import DefinitionCard from './DefinitionCard.js';
import MultipleChoice from './MultipleChoice.js';

class Learn extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  render() {
    let whatToRender = <p>Preparing your terms...</p>
    if(this.props.currentTerm && this.props.wrongAnswers) {
      whatToRender = <div className='term-container'>
                      <DefinitionCard definition={this.props.currentTerm.definition} termType={this.props.currentTerm.type} />
                      <MultipleChoice currentAnswer={this.props.currentTerm.term} wrongAnswers={this.props.wrongAnswers} displayNextTerm={this.props.displayNextTerm} />
                    </div>
    }
    return (
      <div className='learn-container'>
        <hgroup>
          <h1>Match</h1>
          <h4>Get 3 correct and a tree will grow in your forest!</h4>
        </hgroup>
        {whatToRender}
        <h2>{this.props.points} correct answers!</h2>
      </div>
    )
  }
}

export default Learn;