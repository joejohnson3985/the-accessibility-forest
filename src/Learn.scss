@import './_variables';

.learn-container {
  @include center-flex;
  flex-direction: column;
  justify-content: space-around;
  height: 70%;
  width: 65%;
  background-color: rgba(218,100,82, .8);
  padding: 20px;
  border-radius: 5px;
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
  color: #fff;
}

.term-container {
  @include center-flex;
  justify-content: space-around;
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
  background-color: $plain-background;
  color: $primary-color;
  border-radius: 4px;
  padding: 8px;
}