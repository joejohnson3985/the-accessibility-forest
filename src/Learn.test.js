import React from 'react';
import Learn from './Learn';
import { shallow } from 'enzyme';

const mockWrongAnswers = ["<header>", "<article>", "<cite>"]
const mockCurrentTerm = {type: "html tag", term: "<address>", definition: "Indicates that the enclosed HTML provides contact information for a person or people, or for an organization."}

describe('Learn', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <Learn  
          currentTerm={mockCurrentTerm}
          wrongAnswers={mockWrongAnswers}
        />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})

describe('Learn no props', () => {
  const noCurrentTerm = null
  const noWrongAnswers= []
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <Learn  
          currentTerm={noCurrentTerm}
          wrongAnswers={noWrongAnswers}
        />
      )
  });

  it('Should match the snapshot without props', () => {
    expect(wrapper).toMatchSnapshot();
  })
})