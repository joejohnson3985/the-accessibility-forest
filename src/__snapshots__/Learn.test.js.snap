// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn Should match the snapshot 1`] = `
<div
  className="learn-container"
>
  <hgroup>
    <h1>
      Match
    </h1>
    <h4>
      Get 3 correct and a tree will grow in your forest!
    </h4>
  </hgroup>
  <div
    className="term-container"
  >
    <DefinitionCard
      definition="Indicates that the enclosed HTML provides contact information for a person or people, or for an organization."
      termType="html tag"
    />
    <MultipleChoice
      currentAnswer="<address>"
      wrongAnswers={
        Array [
          "<header>",
          "<article>",
          "<cite>",
        ]
      }
    />
  </div>
  <h2>
     correct answers!
  </h2>
</div>
`;

exports[`Learn no props Should match the snapshot without props 1`] = `
<div
  className="learn-container"
>
  <hgroup>
    <h1>
      Match
    </h1>
    <h4>
      Get 3 correct and a tree will grow in your forest!
    </h4>
  </hgroup>
  <p>
    Preparing your terms...
  </p>
  <h2>
     correct answers!
  </h2>
</div>
`;
